dandi/metadata.py

```diff
@@ -71,6 +91,19 @@ def get_metadata(path: Union[str, Path]) -> Optional[dict]:
             lgr.debug("Failed to get metadata for %s: %s", path, exc)
             return None
 
+    # Pretty fragile way to determine that it's not nwb.
+    # Ideally can find something better, like `is_nwb` or `is_bids`.
+    try:
+        h5py.File(path)
```

Comment:
I do not like this one bit.  How would a human looking at a dataset know whether something is BIDS or not?